vendor/github.com/ardanlabs/kit/cfg/cfg.go

```diff
@@ -0,0 +1,298 @@
+package cfg
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+)
+
+// Config is a goroutine safe configuration store, with a map of values
+// set from a config Provider.
+type Config struct {
+	m  map[string]string
+	mu sync.RWMutex
+}
+
+// Provider is implemented by the user to provide the configuration as a map.
+// There are currently two Providers implemented, EnvProvider and MapProvider.
+type Provider interface {
+	Provide() (map[string]string, error)
+}
+
+//==============================================================================
+
+// New populates a new Config from a Provider. It will return an error if there
+// was any problem reading from the Provider.
+func New(p Provider) (*Config, error) {
+	m, err := p.Provide()
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Config{m: m}
+
+	return c, nil
+}
+
+// Log returns a string to help with logging your configuration. It excludes
+// any values whose key contains the string "PASS".
+func (c *Config) Log() string {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	var buf bytes.Buffer
+	for k, v := range c.m {
+		if !strings.Contains(k, "PASS") {
+			buf.WriteString(k + "=" + v + "\n")
+		}
+	}
+
+	return buf.String()
+}
+
+// String returns the value of the given key as a string. It will return an
+// error if key was not found.
+func (c *Config) String(key string) (string, error) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	value, found := c.m[key]
+	if !found {
+		return "", fmt.Errorf("Unknown key %s !", key)
+	}
+
+	return value, nil
+}
+
+// MustString returns the value of the given key as a string. It will panic if
+// the key was not found.
+func (c *Config) MustString(key string) string {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	value, found := c.m[key]
+	if !found {
+		panic(fmt.Sprintf("Unknown key %s !", key))
+	}
+
+	return value
+}
+
+// SetString adds or modifies the configuration for the specified key and
+// value.
+func (c *Config) SetString(key string, value string) {
+	c.mu.RLock()
+	{
+		c.m[key] = value
+	}
+	c.mu.RUnlock()
+}
+
+// Int returns the value of the given key as an int. It will return an error if
+// the key was not found or the value can't be converted to an int.
+func (c *Config) Int(key string) (int, error) {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	value, found := c.m[key]
+	if !found {
+		return 0, fmt.Errorf("Unknown Key %s !", key)
+	}
+
+	iv, err := strconv.Atoi(value)
+	if err != nil {
+		return 0, err
+	}
+
+	return iv, nil
+}
+
+// MustInt returns the value of the given key as an int. It will panic if the
+// key was not found or the value can't be converted to an int.
+func (c *Config) MustInt(key string) int {
+	c.mu.RLock()
```

Comment:
What is the purpose of these Mutexes? Seems like Config is not shared and when you need it you create a new instance via the New method. So every time you use config you get a new instance.